tools/rtpsrelay/RelayHandler.cpp

```diff
@@ -753,21 +711,14 @@ bool SpdpHandler::do_normal_processing(const ACE_INET_Addr& remote,
         const auto pos = guid_addr_set_map_.find(guid);
         if (pos != guid_addr_set_map_.end()) {
           for (const auto& addr : pos->second.addr_set) {
-            venqueue_message(addr, pos->second.stats_reporter, msg, now);
+            vsend(addr, pos->second.stats_reporter, msg, now);
           }
         }
       }
       stats_reporter.max_fan_out(to.size(), now);
     } else {
-      // Forward to all peers except the application participant.
-      for (auto& p : guid_addr_set_map_) {
-        if (p.first != config_.application_participant_guid()) {
-          for (const auto& addr : p.second.addr_set) {
-            venqueue_message(addr, p.second.stats_reporter, msg, now);
-          }
-        }
-      }
-      stats_reporter.max_fan_out(guid_addr_set_map_.size(), now);
+      HANDLER_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: SpdpHandler::do_normal_processing dropping non-directed SPDP message from application participant\n")));
```

Comment:
Based on this change, should the relay's application participant be configured (at least in part) in code instead of relying on the `.ini` file?

Reply:
I was thinking the same thing.  Let's discuss this off-line.